File: 4332-check-good-integer/check-good-integer.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        digit_sum,square_sum=0,0

        while n:
            digit=n%10
            digit_sum+=digit
            square_sum+=digit*digit
            n//=10

        return (square_sum-digit_sum)>=50

File: 4332-check-good-integer/check-good-integer.cpp
class Solution {
public:
    bool checkGoodInteger(int n) {
        int digit_sum=0,square_sum=0;

        while (n>0){
            int digit=n%10;
            digit_sum+=digit;
            square_sum+=digit*digit;
            n/=10;
        }

        return (square_sum-digit_sum)>=50;
    }
};